Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/execution/TestResultScheduler.kt

```diff
@@ -2,57 +2,86 @@ package de.rki.coronawarnapp.coronatest.execution
-import de.rki.coronawarnapp.util.TimeStamper
+import de.rki.coronawarnapp.coronatest.execution.TestResultScheduler.RatPollingMode.DISABLED
+import de.rki.coronawarnapp.coronatest.execution.TestResultScheduler.RatPollingMode.PHASE1
+import de.rki.coronawarnapp.coronatest.worker.PCRTestResultRetrievalWorker
+import de.rki.coronawarnapp.coronatest.worker.RatResultRetrievalWorker
 import de.rki.coronawarnapp.util.coroutine.await
 import de.rki.coronawarnapp.worker.BackgroundConstants
 import de.rki.coronawarnapp.worker.BackgroundWorkHelper
-import de.rki.coronawarnapp.worker.DiagnosisTestResultRetrievalPeriodicWorker
 import kotlinx.coroutines.runBlocking
 import timber.log.Timber
 import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 
 @Reusable
 class TestResultScheduler @Inject constructor(
-    private val tracingSettings: TracingSettings,
     private val workManager: WorkManager,
-    private val timeStamper: TimeStamper,
 ) {
 
-    private suspend fun isScheduled(): Boolean {
-        val workerInfos = workManager.getWorkInfosForUniqueWork(PCR_TESTRESULT_WORKER_UNIQUEUNAME).await()
+    private var ratWorkerMode = DISABLED
+    val ratResultPeriodicPollingMode
+        get() = ratWorkerMode
 
-        return workerInfos.any { it.isScheduled }
-    }
+    private suspend fun isPcrScheduled() =
+        workManager.getWorkInfosForUniqueWork(PCR_TESTRESULT_WORKER_UNIQUEUNAME)
+            .await()
+            .any { it.isScheduled }
+
+    private val WorkInfo.isScheduled: Boolean
+        get() = state == WorkInfo.State.RUNNING || state == WorkInfo.State.ENQUEUED
 
-    fun setPeriodicTestPolling(enabled: Boolean) {
+    fun setPcrPeriodicTestPollingEnabled(enabled: Boolean) {
         if (enabled) {
             // TODO Refactor runBlocking away
-            val isScheduled = runBlocking { isScheduled() }
+            val isScheduled = runBlocking { isPcrScheduled() }
             if (isScheduled) {
                 Timber.tag(TAG).w("Already scheduled, skipping")
                 return
             }
-            Timber.tag(TAG).i("Queueing test result worker (DIAGNOSIS_TEST_RESULT_PERIODIC_WORKER)")
+            Timber.tag(TAG).i("Queueing pcr test result worker (DIAGNOSIS_TEST_RESULT_PERIODIC_WORKER)")
             workManager.enqueueUniquePeriodicWork(
                 PCR_TESTRESULT_WORKER_UNIQUEUNAME,
                 ExistingPeriodicWorkPolicy.REPLACE,
-                buildDiagnosisTestResultRetrievalPeriodicWork()
+                buildPcrTestResultRetrievalPeriodicWork()
             )
         } else {
             Timber.tag(TAG).d("cancelWorker()")
             workManager.cancelUniqueWork(PCR_TESTRESULT_WORKER_UNIQUEUNAME)
         }
     }
 
-    private fun buildDiagnosisTestResultRetrievalPeriodicWork() =
-        PeriodicWorkRequestBuilder<DiagnosisTestResultRetrievalPeriodicWorker>(
-            BackgroundWorkHelper.getDiagnosisTestResultRetrievalPeriodicWorkTimeInterval(),
+    enum class RatPollingMode {
```

Comment:
What do you think about moving each test result specific code into their own class and then just using the `TestResultScheduler` to combine access?

i.e.

```Kotlin
TestResultScheduler constructor(
val pcrScheduler: ....
val raScheduler:....
)
```

Reply:
Hm 50/50 on this, would it be easier to handle or unnecessary abstraction :thinking: , could prevent mistakes of mixing up RA/PCR, what do you think?

Reply:
i like the idea of moving test result specific code to their own class, because there is no shared logic between the two. why would we need combined access?

Reply:
separated the class - how do you like it?

Reply:
I :heart: it